// swiper@11.1.15/shared/utils.js
import { a as c } from "./ssr-window.esm.js";
function y(e) {
  return e === void 0 && (e = ""), e.trim().split(" ").filter((t) => !!t.trim());
}
function x(e) {
  const t = e;
  Object.keys(t).forEach((n) => {
    try {
      t[n] = null;
    } catch {
    }
    try {
      delete t[n];
    } catch {
    }
  });
}
function b(e, t) {
  return t === void 0 && (t = 0), setTimeout(e, t);
}
function M() {
  return Date.now();
}
function S(e) {
  const t = c();
  let n;
  return t.getComputedStyle && (n = t.getComputedStyle(e, null)), !n && e.currentStyle && (n = e.currentStyle), n || (n = e.style), n;
}
function v(e, t) {
  t === void 0 && (t = "x");
  const n = c();
  let r, o, s;
  const l = S(e);
  return n.WebKitCSSMatrix ? (o = l.transform || l.webkitTransform, o.split(",").length > 6 && (o = o.split(", ").map((i) => i.replace(",", ".")).join(", ")), s = new n.WebKitCSSMatrix(o === "none" ? "" : o)) : (s = l.MozTransform || l.OTransform || l.MsTransform || l.msTransform || l.transform || l.getPropertyValue("transform").replace("translate(", "matrix(1, 0, 0, 1,"), r = s.toString().split(",")), t === "x" && (n.WebKitCSSMatrix ? o = s.m41 : r.length === 16 ? o = parseFloat(r[12]) : o = parseFloat(r[4])), t === "y" && (n.WebKitCSSMatrix ? o = s.m42 : r.length === 16 ? o = parseFloat(r[13]) : o = parseFloat(r[5])), o || 0;
}
function p(e) {
  return typeof e == "object" && e !== null && e.constructor && Object.prototype.toString.call(e).slice(8, -1) === "Object";
}
function E(e) {
  return typeof window < "u" && typeof window.HTMLElement < "u" ? e instanceof HTMLElement : e && (e.nodeType === 1 || e.nodeType === 11);
}
function w() {
  const e = Object(arguments.length <= 0 ? void 0 : arguments[0]), t = ["__proto__", "constructor", "prototype"];
  for (let n = 1; n < arguments.length; n += 1) {
    const r = n < 0 || arguments.length <= n ? void 0 : arguments[n];
    if (r != null && !E(r)) {
      const o = Object.keys(Object(r)).filter((s) => t.indexOf(s) < 0);
      for (let s = 0, l = o.length; s < l; s += 1) {
        const i = o[s], u = Object.getOwnPropertyDescriptor(r, i);
        u !== void 0 && u.enumerable && (p(e[i]) && p(r[i]) ? r[i].__swiper__ ? e[i] = r[i] : w(e[i], r[i]) : !p(e[i]) && p(r[i]) ? (e[i] = {}, r[i].__swiper__ ? e[i] = r[i] : w(e[i], r[i])) : e[i] = r[i]);
      }
    }
  }
  return e;
}
function C(e, t, n) {
  e.style.setProperty(t, n);
}
function O(e) {
  let {
    swiper: t,
    targetPosition: n,
    side: r
  } = e;
  const o = c(), s = -t.translate;
  let l = null, i;
  const u = t.params.speed;
  t.wrapperEl.style.scrollSnapType = "none", o.cancelAnimationFrame(t.cssModeFrameID);
  const d = n > s ? "next" : "prev", g = (f, m) => d === "next" && f >= m || d === "prev" && f <= m, h = () => {
    i = (/* @__PURE__ */ new Date()).getTime(), l === null && (l = i);
    const f = Math.max(Math.min((i - l) / u, 1), 0), m = 0.5 - Math.cos(f * Math.PI) / 2;
    let a = s + m * (n - s);
    if (g(a, n) && (a = n), t.wrapperEl.scrollTo({
      [r]: a
    }), g(a, n)) {
      t.wrapperEl.style.overflow = "hidden", t.wrapperEl.style.scrollSnapType = "", setTimeout(() => {
        t.wrapperEl.style.overflow = "", t.wrapperEl.scrollTo({
          [r]: a
        });
      }), o.cancelAnimationFrame(t.cssModeFrameID);
      return;
    }
    t.cssModeFrameID = o.requestAnimationFrame(h);
  };
  h();
}
function P(e, t) {
  t === void 0 && (t = "");
  const n = [...e.children];
  return e instanceof HTMLSlotElement && n.push(...e.assignedElements()), t ? n.filter((r) => r.matches(t)) : n;
}
function j(e, t) {
  const n = t.contains(e);
  return !n && t instanceof HTMLSlotElement ? [...t.assignedElements()].includes(e) : n;
}
function F(e) {
  try {
    console.warn(e);
    return;
  } catch {
  }
}
function _(e, t) {
  t === void 0 && (t = []);
  const n = document.createElement(e);
  return n.classList.add(...Array.isArray(t) ? t : y(t)), n;
}
function A(e, t) {
  const n = [];
  for (; e.previousElementSibling; ) {
    const r = e.previousElementSibling;
    t ? r.matches(t) && n.push(r) : n.push(r), e = r;
  }
  return n;
}
function I(e, t) {
  const n = [];
  for (; e.nextElementSibling; ) {
    const r = e.nextElementSibling;
    t ? r.matches(t) && n.push(r) : n.push(r), e = r;
  }
  return n;
}
function L(e, t) {
  return c().getComputedStyle(e, null).getPropertyValue(t);
}
function W(e) {
  let t = e, n;
  if (t) {
    for (n = 0; (t = t.previousSibling) !== null; )
      t.nodeType === 1 && (n += 1);
    return n;
  }
}
function k(e, t) {
  const n = [];
  let r = e.parentElement;
  for (; r; )
    n.push(r), r = r.parentElement;
  return n;
}
function D(e, t) {
  function n(r) {
    r.target === e && (t.call(e, r), e.removeEventListener("transitionend", n));
  }
  t && e.addEventListener("transitionend", n);
}
function H(e, t, n) {
  const r = c();
  return e[t === "width" ? "offsetWidth" : "offsetHeight"] + parseFloat(r.getComputedStyle(e, null).getPropertyValue(t === "width" ? "margin-right" : "margin-top")) + parseFloat(r.getComputedStyle(e, null).getPropertyValue(t === "width" ? "margin-left" : "margin-bottom"));
}
export {
  k as a,
  _ as c,
  M as d,
  P as e,
  H as f,
  W as h,
  y as i,
  v as j,
  D as k,
  p as l,
  b as n,
  L as p,
  I as q,
  A as r,
  C as s,
  O as t,
  F as u,
  j as v,
  w,
  x
};
