// st-common-ui-vue3/es/components/StPopover/st-popover-sub-cont-pos.js
var i = /* @__PURE__ */ ((t) => (t.LT = "LT", t.LB = "LB", t.RT = "RT", t.RB = "RB", t.TL = "TL", t.TR = "TR", t.BL = "BL", t.BR = "BR", t))(i || {});
const r = {
  // --------------------------------------------------------------------------------------------------- 水平方向 --------
  LT: {
    position: {
      right: "100%",
      top: 0,
      transformOrigin: "right top"
    }
  },
  LB: {
    position: {
      right: "100%",
      bottom: 0,
      transformOrigin: "right bottom"
    }
  },
  RT: {
    position: {
      left: "100%",
      top: 0,
      transformOrigin: "left top"
    }
  },
  RB: {
    position: {
      left: "100%",
      bottom: 0,
      transformOrigin: "left bottom"
    }
  },
  // --------------------------------------------------------------------------------------------------- 垂直方向 --------
  TL: {
    position: {
      bottom: "100%",
      left: 0,
      transformOrigin: "bottom left"
    }
  },
  TR: {
    position: {
      bottom: "100%",
      right: 0,
      transformOrigin: "bottom right"
    }
  },
  BL: {
    position: {
      top: "100%",
      left: 0,
      transformOrigin: "top left"
    }
  },
  BR: {
    position: {
      top: "100%",
      right: 0,
      transformOrigin: "top right"
    }
  }
};
export {
  r as StPopoverSubContPosConst,
  i as StPopoverSubContPosEnum
};
