// st-common-ui-vue3/es/components/StGradientBg1/StGradientBg1.vue2.js
import { defineComponent as s, ref as i, onBeforeMount as c, watch as d, openBlock as u, createElementBlock as p, normalizeStyle as f, unref as a, renderSlot as m } from "vue";
import * as n from "../../node_modules/.pnpm/st-common-ui-utils@0.1.1/node_modules/st-common-ui-utils/es/utils/el-size-util.js";
const z = /* @__PURE__ */ s({
  name: "StGradientBg1",
  __name: "StGradientBg1",
  props: {
    colors: {},
    size: { default: "500%" },
    speed: { default: 2e4 }
  },
  setup(t) {
    const e = t, r = i(e.colors), l = () => {
      !e.colors || !Array.isArray(e.colors) || e.colors.length < 2 ? r.value = ["#2c3e50", "#27ae60", "#2980b9", "#e74c3c", "#8e44ad"] : r.value = e.colors;
    };
    return c(() => {
      l();
    }), d(() => e.colors, () => l(), { deep: !0 }), (o, g) => (u(), p("div", {
      class: "st-gradient-bg-1",
      style: f({
        background: `linear-gradient(135deg, ${r.value.join(",")})`,
        backgroundSize: a(n).elSizePreHandler(o.size),
        animationDuration: a(n).elSizePreHandler(o.speed, "ms")
      })
    }, [
      m(o.$slots, "default", {}, void 0, !0)
    ], 4));
  }
});
export {
  z as default
};
